Paco/assets/js_api_V2/paco_core.js

```diff
@@ -0,0 +1,481 @@
+      actionTriggerId, actionId, actionTriggerSpecId, responses, scheduledTime) {
+    var egName = experimentGroup != null ? experimentGroup.name : "";
+    return  {
+      "experimentId" : experiment.id,
+      "experimentVersion" : experiment.version,
+      "experimentGroupName" : egName,
+      "actionTriggerId" : actionTriggerId,
+      "actionId" : actionId, 
+      "actionTriggerSpecId" : actionTriggerSpecId, 
+      "responseTime" : null, 
+      "scheduledTime" : scheduledTime,
+      "responses" : responses
+    };
+  };
+
+  /*
+   * This function is responsible for creating a response event object for an experimentGroup
+   * @parameters - experiment, experimentGroup, actionTriggerId, actionId, actionTriggerSpecId, scheduledTime
+   * @return - returns a response event object
+   */
+  obj.createResponseEventForExperimentGroup = function(experiment, experimentGroup, 
+      actionTriggerId, actionId, actionTriggerSpecId, scheduledTime) {
+    var inputs = experimentGroup != null ? experimentGroup.inputs : null;
+    var responses = obj.createResponsesForInputs(inputs);
+    
+    return obj.createResponseEventForExperimentWithResponses(experiment, experimentGroup, 
+        actionTriggerId, actionId, actionTriggerSpecId, responses, scheduledTime);
+  };
+  
+  /*
+   * This function is responsible for creating a response event object for an experiment
+   * @parameters - experiment, experimentGroup, scheduledTime
+   * @return - returns a response event object
+   */
+  obj.createResponseEventForExperiment = function(experiment, scheduledTime) {
+    return obj.createResponseEventForExperimentGroup(experiment, null, null, null, null, scheduledTime);
+  };
+
+  obj.answerHas = function(answer, value) {
+    return answer.indexOf(value) != -1;
+  };
+  
+  function isNumeric(num) {
+    return (num >=0 || num < 0);
+  };
+
+  function validNumber(val) {
+    if (!isNumeric(val)) {
+      return false;
+    }
+    try {
+      // is this necessary given the isNumeric test?
+      parseFloat(val);
+      return true;
+    } catch (e) {
+      return false;
+    }     
+  };
+
+  function validValueForResponseType(response) {
+    if (response.responseType === "number") {
+      return validNumber(response.answer);
+    } else {
+      return true;
+    }
+  };
+  
+  valid = function(input, inputHtml, response) { 
+    if ((input.required && inputHtml.element[0].style.display != "none") && (!response.answer || response.answer.length === 0)) {
+      return { "succeeded" : false , "error" : "Response required for " + input.name, "name" : input.name};    
+    } else if (!validValueForResponseType(response)) {
+      return { "succeeded" : false , "error" : "Response required for " + name, "name" : name};    
+    } else {
+      return { "succeeded" : true };
+    }
+  };
+  
+  
+  obj.validate = function(experimentGroup, responseEvent, inputHtmls, errorMarkingCallback) {
```

Comment:
this function obj.validate, should go in the render file as well. (touches html)